bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/connection/ConnectionManager.java

```diff
@@ -0,0 +1,540 @@
+        this.deviceIsConnected = false;
+        this.ipAddress = ipAddress;
+        this.ipPort = ipPort;
+        this.timeout = timeout;
+        this.key = key;
+        this.token = token;
+        this.cloud = cloud;
+        this.deviceId = deviceId;
+        this.version = version;
+        this.promptTone = promptTone;
+        this.lastResponse = new Response(HexFormat.of().parseHex("C00042667F7F003C0000046066000000000000000000F9ECDB"),
+                version, "query", (byte) 0xc0);
+        this.cloudProvider = CloudProviderDTO.getCloudProvider(cloud);
+        this.security = new Security(cloudProvider);
+    }
+
+    private Socket socket = new Socket();
+    private InputStream inputStream = new ByteArrayInputStream(new byte[0]);
+    private DataOutputStream writer = new DataOutputStream(System.out);
+
+    /**
+     * Gets last response
+     * 
+     * @return byte array of last response
+     */
+    public Response getLastResponse() {
+        return this.lastResponse;
+    }
+
+    /**
+     * Validate if String is blank
+     * 
+     * @param str string to be evaluated
+     * @return boolean true or false
+     */
+    public static boolean isBlank(String str) {
+        return str.trim().isEmpty();
+    }
+
+    /**
+     * After checking if the key and token need to be updated (Default = 0 Never)
+     * The socket is established with the writer and inputStream (for reading responses)
+     * The device is considered connected. V2 devices will proceed to send the poll or the
+     * set command. V3 devices will proceed to authenticate
+     */
+    public synchronized void connect() throws MideaConnectionException, MideaAuthenticationException {
+        logger.trace("Connecting to {}:{}", ipAddress, ipPort);
+
+        int maxTries = 3;
+        int retryCount = 0;
+
+        // Open socket
+        // Retry addresses most common wifi connection problems- wait 5 seconds and try again
+        while (retryCount < maxTries) {
+            try {
+                socket = new Socket();
+                socket.setSoTimeout(timeout * 1000);
+                socket.connect(new InetSocketAddress(ipAddress, ipPort), timeout * 1000);
+                break;
+            } catch (IOException e) {
+                retryCount++;
+                if (retryCount < maxTries) {
+                    try {
+                        Thread.sleep(5000);
+                    } catch (InterruptedException ex) {
+                        logger.debug("An interupted error (socket retry) has occured {}", ex.getMessage());
+                    }
+                    logger.debug("Socket retry count {}, IOException connecting to {}: {}", retryCount, ipAddress,
+                            e.getMessage());
+                }
+            }
+        }
+        if (retryCount == maxTries) {
+            deviceIsConnected = false;
+            logger.info("Failed to connect after {} tries. Try again with next scheduled poll", maxTries);
```

Comment:
```suggestion
            logger.debug("Failed to connect after {} tries. Try again with next scheduled poll", maxTries);
```

Reply:
Ok.  Had that for testing, forgot to take out